bugsnag-android-core/src/main/java/com/bugsnag/android/SessionTracker.java

```diff
@@ -92,12 +93,27 @@ Session startNewSession(@NonNull Date date, @Nullable User user,
     }
 
     Session startSession(boolean autoCaptured) {
-        if (client.getConfig().shouldDiscardSession(autoCaptured)) {
+        if (shouldDiscardSession(autoCaptured)) {
             return null;
         }
         return startNewSession(new Date(), client.getUser(), autoCaptured);
     }
 
+    private boolean shouldDiscardSession(boolean autoCaptured) {
+        if (client.getConfig().shouldDiscardSession(autoCaptured)) {
+            return true;
+        } else {
+            if (autoCaptured && currentSession != null) {
+                if (!currentSession.isAutoCaptured() && shouldSuppressFirstAutoSession) {
+                    shouldSuppressFirstAutoSession = false;
+                    return true;
+                }
+            }
+            return false;
+        }
```

Comment:
To avoid possible `NullPointerException`s I think that the `currentSession` should rather be captured here. It's also possible to simplify this `if` statement:

```suggestion
        } else {
            Session existingSession = currentSession;
            if (autoCaptured &&
                    existingSession != null &&
                    !existingSession.isAutoCaptured() &&
                    shouldSuppressFirstAutoSession) {
                    
                shouldSuppressFirstAutoSession = false;
                return true;
            }
        }
        
        return false;
```